File: 2023.2/TEP/ATCODER5/B.cpp
#include <iostream>
#include <climits>

using namespace std;

int minimum_cost(int N, int S, int M, int L) {
    int min_cost = INT_MAX;

    for (int i = 0; i <= N / 6; i++) {
        for (int j = 0; j <= N / 8; j++) {
            int k = (N - i * 6 - j * 8 + 11) / 12;
            k = max(0, k);
            int total_eggs = i * 6 + j * 8 + k * 12;

            if (total_eggs >= N) {
                int cost = i * S + j * M + k * L;
                min_cost = min(cost, min_cost);
            }
        }
    }

    return min_cost;
}

int main() {
    int N, S, M, L;
    cin >> N >> S >> M >> L;

    int result = minimum_cost(N, S, M, L);
    cout << result << endl;

    return 0;
}

File: 2023.2/TEP/ATCODER5/B.py
def minimum_cost(N, S, M, L):
    min_cost = float('inf')

    for i in range(N//6 + 1):
        for j in range(N//8 + 1):
            k = max(0, (N - i * 6 - j * 8 + 11) // 12)
            total_eggs = i * 6 + j * 8 + k * 12

            if total_eggs >= N:
                cost = i * S + j * M + k * L
                min_cost = min(min_cost, cost)

    return min_cost

# Leitura da entrada
N, S, M, L = map(int, input().split())

result = minimum_cost(N, S, M, L)
print(result)
